### ISSUE TITLE
Interface Drupal\config_translation\FormElement\ElementInterface does not exist

### ISSUE THREAD

**Issue by mxr576** [mglaman/phpstan-drupal#131]
There is some kind of weird autoloading issue that I bumped on a project. I get the same error if I run the check on the node module too.

Drupal core: 8.8.8
Composer: **2.0.0-alpha2**
PHP: 7.2
```
mglaman/drupal-check                           1.1.2               CLI tool for running checks on a Drupal code base
mglaman/phpstan-drupal                         0.12.3              Drupal extension and rules for PHPStan
phpstan/phpdoc-parser                          0.3.5               PHPDoc parser with support for nullable, intersection and generic types
phpstan/phpstan                                0.12.25             PHPStan - PHP Static Analysis Tool
phpstan/phpstan-deprecation-rules              0.12.4              PHPStan rules for detecting usage of deprecated classes, methods, properties, constants and traits.
```

```sh
 scripts/drupal-check-wrapper.sh web/modules/drupal_module/ 
/mnt/files/local_mount/drupal/modules/drupal_module/drupal_module.module
/mnt/files/local_mount/drupal/modules/drupal_module/tests/src/Functional/AccessTest.php
[30-Jun-2020 21:43:50 Australia/Sydney] PHP Fatal error:  Uncaught ReflectionException: Interface Drupal\config_translation\FormElement\ElementInterface does not exist in /mnt/files/local_mount/drupal/modules/drupal_module/build/vendor/mglaman/phpstan-drupal/src/Rules/Drupal/GlobalDrupalDependencyInjectionRule.php:47
Stack trace:
#0 /mnt/files/local_mount/drupal/modules/drupal_module/build/vendor/mglaman/phpstan-drupal/src/Rules/Drupal/GlobalDrupalDependencyInjectionRule.php(47): ReflectionClass->implementsInterface('Drupal\\config_t...')
#1 phar:///mnt/files/local_mount/drupal/modules/drupal_module/build/vendor/phpstan/phpstan/phpstan/src/Analyser/FileAnalyser.php(69): PHPStan\Rules\Drupal\GlobalDrupalDependencyInjectionRule->processNode(Object(PhpParser\Node\Expr\StaticCall), Object(PHPStan\Analyser\MutatingScope))
#2 phar:///mnt/files/local_mount/drupal/modules/drupal_module/build/vendor/phpstan/phpstan/phpstan/src/Analyser/NodeScopeResolver.php(300): PHPStan\Analyser\FileAnalyser->PHPStan\Analyser\{closure}(Object(PhpParser\Node\Expr\StaticCall), Object(PHPStan\Analyser\MutatingScope))
#3 phar:///mnt/files/local_mount/drupal/modules/drupal_module/build/vendor/phpstan/phpstan/phpstan/src/Analyser/NodeScopeResolver.php(1481): PHPStan\Analyser\NodeScopeResolver::PHPStan\Analyser\{closure}(Object(PhpParser\Node\Expr\StaticCall), Object(PHPStan\Analyser\MutatingScope))
```

**Comment by mxr576** [comment]
An interesting fact that if I remove [this interface from whitelist](https://github.com/mglaman/phpstan-drupal/blob/0.12.4/src/Rules/Drupal/GlobalDrupalDependencyInjectionRule.php#L40) then Drupal Check does not fail but it complains about:

```
Line   src/Access/NodeAddAccessCheck.php                                
 ------ ----------------------------------------------------------------- 
         Class Drupal\node\Access\NodeAddAccessCheck was not found while  
         trying to analyse it - autoloading is probably not configured    
         properly.                                                        
         💡 Learn more at https://phpstan.org/user-guide/autoloading       
 ------ ----------------------------------------------------------------- 

 ------ --------------------------------------------------------------------- 
  Line   src/Access/NodePreviewAccessCheck.php                                
 ------ --------------------------------------------------------------------- 
         Class Drupal\node\Access\NodePreviewAccessCheck was not found while  
         trying to analyse it - autoloading is probably not configured        
         properly.                                                            
         💡 Learn more at https://phpstan.org/user-guide/autoloading           
 ------ --------------------------------------------------------------------- 

 ------ ---------------------------------------------------------------------- 
  Line   src/Access/NodeRevisionAccessCheck.php                                
 ------ ---------------------------------------------------------------------- 
         Class Drupal\node\Access\NodeRevisionAccessCheck was not found while  
         trying to analyse it - autoloading is probably not configured         
         properly.                                                             
         💡 Learn more at https://phpstan.org/user-guide/autoloading            
 ------ ---------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------- 
  Line   src/Cache/NodeAccessGrantsCacheContext.php                           
 ------ --------------------------------------------------------------------- 
         Class Drupal\node\Cache\NodeAccessGrantsCacheContext was not found   
         while trying to analyse it - autoloading is probably not configured  
         properly.                                                            
         💡 Learn more at https://phpstan.org/user-guide/autoloading           
 ------ --------------------------------------------------------------------- 

```

**Comment by mglaman** [comment]
@mxr576 can you provide code samples that help reproduce because I'm not sure why it cannot find that interface, it is still in the code base: https://git.drupalcode.org/project/drupal/-/blob/8.8.x/core/modules/config_translation/src/FormElement/ElementInterface.php

**Comment by mxr576** [comment]
I'll try to figure out sometime what is the minimum code that needs to  reproduce this, currently we stuck with Composer 2 migration. I also confirmed that the file exists before I opened this issue. Something weird is going on... 99% of test files passes code checks. In both cases we have symlinked components, including Drupal core.

**Comment by Frank-Landry** [comment]
@mglaman @mxr576 I ran into this exact same issue trying to scan my custom modules. This happened if I ran phpstan-drupal from the root directory or inside the module root. To solve the issue for me, I removed all phpstan-drupal entries from composer.json, then re-imported them with composer require --dev option. In the phpstan.neon file that I created at the root level, I also had to specify the drupal_root: entry and provide it a valid path to the drupal root. I would try that and see if it works.
